Application.mk

```diff
@@ -151,6 +151,7 @@ PROGLIST := $(addprefix $(BINDIR)$(DELIM),$(PROGLIST))
 PROGOBJ := $(MAINOBJ)
 
 $(PROGLIST): $(MAINOBJ)
+	$(Q) mkdir -p $(BINDIR)
```

Comment:
1) mkdir should not be used explicitly.  This is incompatible with Window native builds.  You should define something.
2) If my recollection is correct, mkdir -p does not work on some Unix platforms, I think macOS.

Reply:
Certainly this is a big risk because I don't see any reference to mkdir in any other Make* or *.mk file.  We need to be careful here.